Bottom: 0beb8d73e0c9298086cfddb6401ac4226b1a38dd
Top:    1795558fe8ac56f362f38a5f6818bcea05667bf5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-12-27 18:12:32 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 10796fc..63d7ca4 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1187,7 +1187,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
